src/main/java/org/opensearch/sdk/handlers/ExtensionActionRequestHandler.java

```diff
@@ -0,0 +1,140 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.handlers;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionResponse;
+import org.opensearch.action.ActionType;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.extensions.action.ExtensionActionRequest;
+import org.opensearch.extensions.action.RemoteExtensionActionResponse;
+import org.opensearch.sdk.SDKClient;
+import org.opensearch.sdk.SDKTransportService;
+import org.opensearch.sdk.action.SDKActionModule;
+
+/**
+ * This class handles a request from OpenSearch from another extension's {@link SDKTransportService#sendProxyActionRequest()} call.
+ */
+public class ExtensionActionRequestHandler {
+    private static final Logger logger = LogManager.getLogger(ExtensionActionRequestHandler.class);
+
+    final SDKActionModule sdkActionModule;
+    final SDKClient client;
+
+    /**
+     * Instantiate this handler
+     *
+     * @param sdkClient An initialized SDKClient with the registered actions
+     * @param sdkActionModule An initialized SDKActionModule with the registered actions
+     */
+    public ExtensionActionRequestHandler(SDKClient sdkClient, SDKActionModule sdkActionModule) {
+        this.sdkActionModule = sdkActionModule;
+        this.client = sdkClient;
+    }
+
+    /**
+     * Handles a request from OpenSearch to execute a TransportAction on the extension.
+     *
+     * @param request The request to execute
+     * @return The response from the TransportAction
+     */
+    public RemoteExtensionActionResponse handleExtensionActionRequest(ExtensionActionRequest request) {
+        logger.debug("Received request to execute action [" + request.getAction() + "]");
+        final RemoteExtensionActionResponse response = new RemoteExtensionActionResponse(false, new byte[0]);
+
+        // Find matching ActionType instance
+        Optional<? extends ActionType<? extends ActionResponse>> optionalAction = sdkActionModule.getActions()
+            .values()
+            .stream()
+            .map(h -> h.getAction())
+            .filter(a -> a.getClass().getName().equals(request.getAction()))
+            .findAny();
+        if (optionalAction.isEmpty()) {
+            response.setResponseBytesAsString("No action [" + request.getAction() + "] is registered.");
+            return response;
+        }
```

Comment:
I would like this to be a function of `SDKActionModule`. Sounds more like a utility.

Reply:
See [this comment](https://github.com/opensearch-project/opensearch-sdk-java/pull/588#discussion_r1146694726). This has been replaced by a function that already existed on `SDKClient`.